src/components/BaseCommentForm.js

```diff
@@ -320,347 +385,382 @@ export class BaseCommentForm extends React.Component {
+          >
+            <div>
+              <label htmlFor="commentMapAddress">
+                <FormattedMessage id="commentMapTitle" />
               </label>
             </div>
-            <span style={{fontSize: 13, marginTop: 20}}><FormattedMessage id="multipleImages"/></span>
-          </FormGroup>
-          { this.renderNameFormForUser() }
-          <div className="comment-buttons clearfix">
-            <Button
-              bsStyle="default"
-              onClick={this.toggle.bind(this)}
-            >
-              <FormattedMessage id="cancel"/>
-            </Button>
-            <Button
-              bsStyle="primary"
-              aria-disabled={this.hasFormErrors()}
-              className={this.hasFormErrors() ? 'disabled' : null}
-              onClick={this.submitComment.bind(this)}
-            >
-              <FormattedMessage id="submit"/>
-            </Button>
+            <FormattedMessage id="commentMapInstructions">
+              {(instr) => <span style={{ fontSize: 13 }}>{instr}</span>}
+            </FormattedMessage>
+            <div className="map-padding">
+              <CommentFormMap
+                center={getMapCenter()}
+                mapBounds={localization.mapBounds || null}
+                mapTileUrl={getMapContrastTiles()}
+                onDrawCreate={onDrawCreate}
+                onDrawDelete={onDrawDelete}
+                contents={getMapBorder()}
+                tools={section.commenting_map_tools}
+                language={language}
+              />
+            </div>
+            <FormGroup>
+              <ControlLabel htmlFor="map_text">
+                <FormattedMessage id="commentMapAdditionalInfo" />
+              </ControlLabel>
+              <FormControl
+                id="map_text"
+                type="text"
+                value={formData.mapCommentText}
+                onChange={handleMapTextChange}
+                maxLength={128}
+              />
+            </FormGroup>
           </div>
-          <CommentFormErrors
-            commentRequiredError={this.state.commentRequiredError}
-            commentOrAnswerRequiredError={this.state.commentOrAnswerRequiredError}
-            imageTooBig={this.state.imageTooBig}
-          />
-          <CommentDisclaimer/>
-        </form>
-      </div>
-    );
-  }
-}
+        )}
+
+        <div className="comment-form__selected-images">
+          {formData.imageTooBig ? (
+            <div className="comment-form__image-too-big">
+              <FormattedMessage id="imageSizeError" />
+            </div>
+          ) : (
+            formData.images.map((image, key) => (
+              <img
+                style={{ marginRight: 10 }}
+                alt=""
+                src={image.image}
+                width={image.width < 100 ? image.width : 100}
+                height={image.height < 100 ? image.width : 100}
+                key={key + Math.random()} //eslint-disable-line
```

Comment:
OK this is old code, but how about using image.image (base64 url string) + index number as key? Not foolproof.. alternatively generate random uuids when the images are processed in handleChange ?

Reply:
https://github.com/City-of-Helsinki/kerrokantasi-ui/pull/958/commits/fe0e02c46761f286c0a3c2455f5dc45ba13c279a